codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt

```diff
@@ -191,7 +198,10 @@ class PresignableGetIntegration(private val presignedOperations: Map<String, Set
         }
     }
 
-    private fun overrideHttpMethodWithGet(): String {
-        return "get"
+    private fun overrideHttpMethod(operation: OperationShape): String {
+        return when (operation.id.toString()) {
+            "com.amazonaws.s3#PutObject" -> "put"
+            else -> "get"
+        }
     }
```

Comment:
is the real edge case here polly? The HTTP method for all S3 operations is preserved AFAIK?

Reply:
yes edge case is polly so this is basically just for polly I should change this to reflect that, good call out

Reply:
actually requires changing a lot more code. leaving for now but its a good call out for the future.